apps/src/gamelab/GameLab.js

```diff
@@ -42,15 +43,10 @@ var GameLab = function () {
   this.eventHandlers = {};
   this.Globals = {};
   this.currentCmdQueue = null;
-  this.p5 = null;
-  this.p5decrementPreload = null;
-  this.p5eventNames = [
-    'mouseMoved', 'mouseDragged', 'mousePressed', 'mouseReleased',
-    'mouseClicked', 'mouseWheel',
-    'keyPressed', 'keyReleased', 'keyTyped'
-  ];
-  this.p5specialFunctions = ['preload', 'draw', 'setup'].concat(this.p5eventNames);
-
+  this.drawInProgress = false;
+  this.setupInProgress = false;
```

Comment:
Just a thought: It seems unlikely to happen, but would it be harmful for both of these to be `true` at the same time?  If so, would it be better to model them as a single state?